core/backend.go

```diff
@@ -61,11 +61,7 @@ type Backend interface {
 	// ID returns the validator's ID
 	ID() []byte
 
-	// MaximumFaultyNodes returns the maximum number of faulty nodes based
-	// on the validator set.
-	MaximumFaultyNodes() uint64
-
-	// Quorum returns what is the quorum size for the
-	// specified block height.
-	Quorum(blockHeight uint64) uint64
+	// HasQuorum returns true if the quorum is reached
+	// for the specified block height.
+	HasQuorum(blockNumber uint64, messages []*proto.Message, msgType proto.MessageType) bool
```

Comment:
It is a bit surprisingly to me that we need `msgType` as a parameter here...

Based on https://github.com/0xPolygon/go-ibft/pull/39/commits/83353bef1c07c08cb5fea7f4302d64bcd449e870, I've understood that sometimes in the tests, messages were empty. Although in that case, we should return `false` immediately, no need to extract message type. 

Btw, do we have situations when `messages` slice contains mixed message types, so that we cannot deduce with certainty in which case it should fall and what should qurum check look like, so that we need `msgType` parameter, because of such use case?

Reply:
There's one case @Stefan-Ethernal in single node, the subscription for preprepare messages is not getting the actual message, it's just being called with empty message, we need to check that's that message type and allow the quorum.

As far as I've seen there's only one case where we have mixed message types, here https://github.com/0xPolygon/go-ibft/blob/feature/EVM-142-move-quorum-to-backend/core/ibft.go#L1047

And as I understand they need to come together always so I chose `PREPARE` as the message type.